experimental/dds/tree/src/id-compressor/idCompressorLogger.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ITelemetryBaseEvent,
+	ITelemetryErrorEvent,
+	ITelemetryGenericEvent,
+	ITelemetryLogger,
+	ITelemetryPerformanceEvent,
+} from '@fluidframework/common-definitions';
+
+export class IdCompressorLogger implements ITelemetryLogger {
+	public constructor(private readonly logger: ITelemetryLogger, private readonly sendEvents: boolean) {}
```

Comment:
Nice! My last recommendation would be to let you specify the percentage in the constructor here, maybe even have it default to 5% so that you don't have to do anything at the call site in SharedTree.ts (except construct the logger). Normally, I'm all for injecting behavior, injecting policy, and having classes be as abstract and generic as reasonable, but... since we already went the route with this particular class of "make it very specific to IdCompressor", I'd say just throw as much code inside of it as possible so that the sampling kludginess is all in one place. I guess what I'm saying is, either let's make it truly generic (`RandomChangeDisabledLogger`) and have it live in common, or let's make it as ungeneric as possible and have it live in IdCompressor, stuffing all the logic related to this sampling stuff inside of it. For the latter case I'd even go so far as to say let's not export the IdCompressorLogger at all and instead export a function `createIdCompressorWithThrottledLogging` that hooks it up under the hood. Then the only surface exposure for this whole thing is one function.

These are nitpicks at this point; do what you think is best. I just wanted to give some ideas and rationale for why you might choose some approaches over others.

Reply:
FWIW, we will want very similar logic in the other compressor. Not sure that changes things much, but worth mentioning.

Reply:
I just found `TelemetryNullLogger` in `telemetry-utils` and I think we might not need this class at all. How about a function `createThrottledIdCompressorLogger(logger, percentage)` that passes in the `TelemetryNullLogger` if we're not sampled?

Reply:
Love it, nice find! This is way better.